addon/utils/serialize-and-push.js

```diff
@@ -0,0 +1,10 @@
+import { isArray } from '@ember/array';
+
+export default function(options, response) {
+  let modelName = this.constructor.modelName || this.constructor.typeKey;
+  let modelClass = this.store.modelFor(modelName);
+  let serializer = this.store.serializerFor(modelName);
+  let normalized = isArray(response.data) ? serializer.normalizeArrayResponse(this.store, modelClass, response) :
```

Comment:
This assumes that all consumers use JSON-API for their backend -- this library currently makes no assumptions about API contract.

I'd be ok with you adding in some smart JSON-API behavior like this, but you'd have to only do it if you can conclusively identify that you're receiving an appropriate payload. I suggest checking for the presence of [the mandatory `jsonapi` top-level object](https://jsonapi.org/format/#document-jsonapi-object).

Reply:
Good idea, I've added a check for this.